app/authorize.html

```diff
@@ -2,57 +2,66 @@
 <html lang="en">
 <body>
 <script>
-
 function getCookie(cname) {
-
     var name = cname + "=";
     var ca = document.cookie.split(';');
-
     for(var i=0; i<ca.length; i++) {
         var c = ca[i].trim();
         if (c.indexOf(name)==0) return c.substring(name.length,c.length);
     }
     return "";
 }
-
 function setCookie(cname, cvalue, exdays)
 {
     if(cvalue===null || cvalue === undefined)
         exdays = -365;
-
     var parts = [cname + "=" + (cvalue||"")];
-
     if(exdays) {
         var d = new Date();
         if(typeof exdays == 'string' && /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{1,8}Z/.test(exdays))
             d = new Date(exdays)
         else      
             d.setDate(d.getDate()+exdays);
-
         parts.push("expires="+d.toGMTString())
     }    
-
     parts.push("path=/");
-
     document.cookie =  parts.join("; ");
 }
-
-var allowedHosts = [ 'www','absch', 'bch', 'bch-training', 'bch-demo', 'chm', 'eunomia', 'inde', 'lifeweb', 'localhost', 'training-absch', 'dev-absch', 'dev-chm', 'chm-training', 'oasis' ];
-
+var trustedHosts = [
+    [/^staging\.cbd\.int$/i, /.*\.staging\.cbd\.int$/i],
+    [/^cbddev\.xyz$/i, /.*\.cbddev\.xyz$/i],
+    [/^cbd\.int$/i, /.*\.cbd\.int$/i]
+];
 function receiveMessage(event) {
-
     var secureDomain = document.location.hostname.replace(/[^\.]+\./, '');
     var origin = event.origin;
-
     if(origin=='http://localhost' || /^http:\/\/localhost:/.test(origin)) origin = 'https://localhost.'+secureDomain;
-
     var originDomain = origin.match(/(?:^(https)\:\/\/)([^\.]+)\.([^\/]+)/)[3];
     var originHost   = origin.match(/(?:^(https)\:\/\/)([^\.]+)\.([^\/]+)/)[2];
-
-    if(originDomain==secureDomain && ~allowedHosts.indexOf(originHost)) {
-
+    
+    for (var j = 0; j < trustedHosts.length; j++) {
+        let exitFor = false;
+        let domainMatched = false;
+        let originMatched = false;
+        for (var i = 0; i < trustedHosts[j].length; i++) {
+            var trustedHost = trustedHosts[j][i];
+            // If the secure domain was matched then the origin domain needs to be on same domain, case for 4 level domains
+            // eg absch.staging.cbd.int
+            domainMatched = domainMatched || trustedHost.test(secureDomain);
+            originMatched = originMatched || trustedHost.test(originDomain);
+            if(domainMatched || originMatched)
+                exitFor = true;
+            
+            if(domainMatched && originMatched){
+                isTrustedDomain = true;
+                break;
+            }
+        }
+        if(exitFor)
```

Comment:
https://stackoverflow.com/questions/183161/whats-the-best-way-to-break-from-nested-loops-in-javascript